src/corespotlight.cs

```diff
@@ -1105,4 +1117,123 @@ interface CSImportExtension : NSExtensionRequestHandling
+		Action<NSArray<CSSuggestion>> FoundSuggestionsHandler { get; set; }
+
+		[Export ("start")]
+		void Start ();
+
+		[Export ("cancel")]
+		void Cancel ();
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (CSSearchQueryContext))]
+	interface CSUserQueryContext
+	{
+		[Static]
+		[Export ("userQueryContext")]
+		CSUserQueryContext UserQueryContext { get; }
+
+		[Static]
+		[Export ("userQueryContextWithCurrentSuggestion:")]
+		CSUserQueryContext Create ([NullAllowed] CSSuggestion currentSuggestion);
+
+		[Export ("maxSuggestionCount")]
+		nint MaxSuggestionCount { get; set; }
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (CSUserQueryContext))]
+	interface CSTopHitQueryContext
+	{
+		[Static]
+		[Export ("topHitQueryContext")]
+		CSTopHitQueryContext TopHitQueryContext { get; }
+
+		[Static]
+		[Export ("topHitQueryContextWithCurrentSuggestion:")]
+		CSTopHitQueryContext Create ([NullAllowed] CSSuggestion currentSuggestion);
+
+		[Export ("maxItemCount")]
+		nint MaxItemCount { get; set; }
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof(CSUserQuery))]
+	[DisableDefaultCtor]
+	interface CSTopHitQuery
+	{
+		[Export ("initWithUserQueryString:queryContext:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] string userQueryString, CSTopHitQueryContext queryContext);
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (NSObject))]
+	interface CSSuggestion : NSSecureCoding, NSCopying
+	{
+
+		[Field ("CSSuggestionHighlightAttributeName")]
+		NSString HighlightAttributeName { get; }
+
+		[Export ("localizedAttributedSuggestion")]
+		NSAttributedString LocalizedAttributedSuggestion { get; }
+
+		[Export ("suggestionKind")]
+		CSSuggestionKind SuggestionKind { get; }
+
+		[Export ("compare:")]
+		NSComparisonResult Compare (CSSuggestion other);
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (NSObject))]
+	interface CSSearchQueryContext : NSSecureCoding, NSCopying
+	{
+		[NullAllowed, Export ("fetchAttributes", ArgumentSemantic.Strong)]
+		string[] FetchAttributes { get; set; }
+
+		[NullAllowed, Export ("protectionClasses", ArgumentSemantic.Strong)]
+		string[] ProtectionClasses { get; set; }
+
+		[NullAllowed, Export ("filterQueries", ArgumentSemantic.Copy)]
+		string[] FilterQueries { get; set; }
+
+		[NullAllowed, Export ("keyboardLanguage", ArgumentSemantic.Strong)]
+		string KeyboardLanguage { get; set; }
+
+		[Export ("sourceOptions", ArgumentSemantic.Assign)]
+		CSSearchQuerySourceOptions SourceOptions { get; set; }
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[Native]
+	public enum CSSearchQuerySourceOptions : long
+	{
+		Default = 0,
+		AllowMail = 1L << 0,
+	}
+
+	[NoTV, Mac (10,13), iOS (16,0), MacCatalyst (16,0)]
+	[Native]
+	public enum CSSuggestionKind : long
+	{
+		None,
+		Custom,
+		Default,
+	}
```

Comment:
Nit: Not a fan of having these at the bottom of the file, we usually put them at the top but I won't die on a hill for this.

Reply:
There are no enums in this file but this ones ;) I can move them, I'm not to bordered to be honest.